registry/registry_test.go

```diff
@@ -0,0 +1,57 @@
+package registry
+
+import (
+	"golang.org/x/net/context"
+	"testing"
+)
+
+func buildAnonymousDockerHub(t *testing.T, ctx context.Context) *Registry {
+	r, err := New(ctx, "https://registry-1.docker.io/", "", "")
+	if err != nil {
+		t.Error(err)
+	}
+	return r
+}
+
+func TestTags(t *testing.T) {
+	ctx, cancel := context.WithCancel(context.Background())
```

Comment:
not sure I understand why you're bothering to WithCancel in each test if you're not actually canceling?  (vs just using context.Background).

Nice to see tests though. Hopefully Docker registry is stable enough to keep them from being flaky :)